src/Runner.Worker/Worker.cs

```diff
@@ -156,8 +156,9 @@ private void InitializeSecretMasker(Pipelines.AgentJobRequestMessage message)
                     HostContext.SecretMasker.AddValue(value);
 
                     // Also add each individual line. Typically individual lines are processed from STDOUT of child processes.
-                    var split = value.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
-                    foreach (var item in split)
+                    var auxiliarySecrets = value.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries)
```

Comment:
you can also get the `DerivedSecretRecommendedMinimumLength` from something inside the message, so you can configure the value from the service via feature flag/registry setting instead of a hardcode `3`